examples/step-68/step-68.cc

```diff
@@ -0,0 +1,810 @@
+
+    // This example does not use adaptive refinement, therefore every cell
+    // should have the status CELL_PERSIST. However this function can also
+    // be used to distribute load during refinement, therefore we consider
+    // refined or coarsened cells as well.
+    if (status == parallel::distributed::Triangulation<dim>::CELL_PERSIST ||
+        status == parallel::distributed::Triangulation<dim>::CELL_REFINE)
+      {
+        const unsigned int n_particles_in_cell =
+          particle_handler.n_particles_in_cell(cell);
+        return n_particles_in_cell * particle_weight;
+      }
+    else if (status == parallel::distributed::Triangulation<dim>::CELL_COARSEN)
+      {
+        unsigned int n_particles_in_cell = 0;
+
+        for (unsigned int child_index = 0;
+             child_index < GeometryInfo<dim>::max_children_per_cell;
+             ++child_index)
+          n_particles_in_cell +=
+            particle_handler.n_particles_in_cell(cell->child(child_index));
+
+        return n_particles_in_cell * particle_weight;
+      }
+
+    Assert(false, ExcInternalError());
+    return 0;
+  }
+
+  // @sect4{Particles generation}
+
+  // This function generates the tracer particles and the background
+  // triangulation on which these particles evolve.
+  template <int dim>
+  void ParticleTracking<dim>::particles_generation()
+  {
+    // We create an hyper_cube triangulation which we globally define. This
+    // triangulation englobes the full trajectory of the particles.
+    GridGenerator::hyper_cube(background_triangulation, 0, 1);
+    background_triangulation.refine_global(par.fluid_refinement);
+
+    // In order to consider the particles when repartitioning the triangulation
+    // the algorithm needs to know three things:
+    //
+    // 1. How much weight to assign to each cell (how many particles are in
+    // there)
+    // 2. How to pack the particles before shipping data around
+    // 3. How to unpack the particles after repartitioning
+    //
+    // We attach the correct functions to the signals inside
+    // parallel::distributed::Triangulation. These signal will be called every
+    // time the repartition() function is called. These connections only need to
+    // be created once, so we might as well have set them up in the constructor
+    // of this class, but for the purpose of this example we want to group the
+    // particle related instructions.
+    background_triangulation.signals.cell_weight.connect(
+      [&](
+        const typename parallel::distributed::Triangulation<dim>::cell_iterator
+          &cell,
+        const typename parallel::distributed::Triangulation<dim>::CellStatus
+          status) -> unsigned int { return this->cell_weight(cell, status); });
+
+    background_triangulation.signals.pre_distributed_repartition.connect(
+      std::bind(
+        &Particles::ParticleHandler<dim>::register_store_callback_function,
+        &particle_handler));
```

Comment:
Sorry for my lack of knowledge, but how can I do that?

Reply:
I think this could work (give it a shot):
```
background_triangulation.signals.pre_distributed_repartition.connect(
  [particle_handler]() { particle_handler.register_store_callback_function(); });
```

Reply:
Sorry for pestering you with this, but I must acknowledge that my knowledge of lambda functions is very limited here. I get the following error message trying your solution : 

`home/blaisb/work/dealii/dealii/dealii/examples/step-68/step-68.cc: In member function ‘void Step68::ParticleTracking<dim>::generate_particles()’:
/home/blaisb/work/dealii/dealii/dealii/examples/step-68/step-68.cc:419:9: error: capture of non-variable ‘Step68::ParticleTracking<dim>::particle_handler’
  419 |       [&particle_handler]() {
      |         ^~~~~~~~~~~~~~~~
/home/blaisb/work/dealii/dealii/dealii/examples/step-68/step-68.cc:278:47: note: ‘dealii::Particles::ParticleHandler<dim> Step68::ParticleTracking<dim>::particle_handler’ declared here
  278 |     Particles::ParticleHandler<dim>           particle_handler;
      |                                               ^~~~~~~~~~~~~~~~
/home/blaisb/work/dealii/dealii/dealii/examples/step-68/step-68.cc: In lambda function:
/home/blaisb/work/dealii/dealii/dealii/examples/step-68/step-68.cc:420:9: error: ‘this’ was not captured for this lambda function
  420 |         particle_handler->register_store_callback_function();
      |         ^~~~~~~~~~~~~~~~
`